controllers/onload_controller.go

```diff
@@ -470,11 +564,14 @@ func (r *OnloadReconciler) handleNodeUpdate(ctx context.Context, onload *onloadv
 		return res, err
 	}
 
-	// Remove kmm label
-	labelName := kmmLabelName(onload.Name, onload.Namespace)
-	err = r.deleteLabelFromNode(ctx, node, labelName)
+	// Remove kmm labels
+	err = r.deleteLabelFromNode(ctx, node, kmmOnloadLabelName(onload.Name, onload.Namespace))
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.deleteLabelFromNode(ctx, node, kmmSFCLabelName(onload.Name, onload.Namespace))
```

Comment:
You should try to remove the sfc module first, then the onload label.
If there is an error here when the reconciler is requeued it won't try to remove the lingering sfc kmm label. (The onload kmm label is used to check whether the onload version on a node is correct).

Reply:
Well spotted!

I've added a comment too.